### ISSUE TITLE
"id" prefix to a project name kills any evaluation of the command

### ISSUE THREAD

**Issue by xyank-github** [GothenburgBitFactory/taskwarrior#2351]
* What command(s) did you run?
```
% task rc.data.location=~/.task add "Test"
Using alternate data.location /home/user/.task
Created task 1.
Configuration override rc.data.location:~/.task
The project 'inbox' has changed.  Project 'inbox' is 0% complete (1 task remaining).
% task rc.data.location=~/.task mod 1 project:ideas
Using alternate data.location /home/user/.task
Configuration override rc.data.location:~/.task
The expression could not be evaluated.
```
* What did you expect to happen?
Task 1 should move to project "ideas"

* What actually happened?
Nothing

* Paste the output of the ```task diag``` command.
```
% task diag

task 2.5.1
   Platform: Linux

Compiler
    Version: 7.2.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      CMake: 3.9.5
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.5.8
 Build type: None

Configuration
       File: /home/user/.taskrc (found), 2106 bytes, mode 100644
       Data: /home/user/.data/.taskdb (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
    $EDITOR: vim
     Server:
      Trust: strict
Certificate: , not readable, 0 bytes
        Key: , not readable, 0 bytes
    Ciphers: NORMAL
      Creds:

Hooks
     System: Enabled
   Location: /home/user/.data/.taskdb/hooks
     Active: on-add-pirate         (executable)
             on-exit.commit.py     (executable)
             on-modify-pirate      (executable)
   Inactive:

Tests
      $TERM: tmux-256color (323x61)
       Dups: Scanned 466 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 466 tasks for broken references:
             No broken references found
```

This clearly is an issue with the "id" part of the project name: if I use the project name "_ideas", there is no issue. The same thing occurs for adding tasks too.
```
% task rc.data.location=~/.task mod 1 project:idoienoie
Using alternate data.location /home/user/.task
Configuration override rc.data.location:~/.task
The expression could not be evaluated.
% task rc.data.location=~/.task mod 1 project:ie
Modifying task 1 'Test'.
Modified 1 task.
Using alternate data.location /home/user/.task
Configuration override rc.data.location:~/.task
The project 'inbox' has changed.  Project 'inbox' is 0% complete (0 of 0 tasks remaining).
The project 'ie' has changed.  Project 'ie' is 0% complete (1 task remaining).
```

**Comment by tbabej** [comment]
@kwoot I'm sure this one was reported in various forms (maybe with referece to other task attributes), but the mechanism should be the same. A bit of issue duplication detective work right here :male_detective:

**Comment by tbabej** [comment]
@xyank-github This is actually fixed on 2.5.2. Installing from source from the 2.5.2 branch should help you avoid this issue. Thanks for reporting!

**Comment by tbabej** [comment]
Bookeeping-wise, this is a duplicate of #1826.
